tests/integration/Models/UserFlagTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace MailPoet\Test\Models;
+
+use MailPoet\Models\UserFlag;
+
+class UserFlagTest extends \MailPoetTest {
+  
+  function testItCanCreateOrUpdate() {
+    expect(UserFlag::count())->equals(0);
+
+    $created_flag = UserFlag::createOrUpdate([
+      'user_id' => 3,
+      'name' => 'first_flag',
+      'value' => 'foo'
+    ]);
+    expect($created_flag->id > 0)->true();
+    expect($created_flag->getErrors())->false();
+    expect(UserFlag::count())->equals(1);
+
+    $updated_flag = UserFlag::createOrUpdate([
+      'user_id' => 3,
+      'name' => 'first_flag',
+      'value' => 'bar'
+    ]);
+    expect($updated_flag->id)->equals($created_flag->id);
+    expect($updated_flag->value)->equals('bar');
+    expect(UserFlag::count())->equals(1);
+
+    $other_flag = UserFlag::createOrUpdate([
+      'user_id' => 3,
+      'name' => 'second_flag',
+      'value' => 'bar'
+    ]);
+    expect($other_flag->id != $created_flag)->true();
+    expect(UserFlag::count())->equals(2);
+  }
+
+  function _after() {
```

Comment:
This should be `_before`, not `_after`. That way you don't need to have an assertion at the beginning of the test.

Reply:
But I still need to clean the DB after the test, otherwise there may be a conflict with some other test.
Do you suggest to add `_before` to clean DB before test too?

Reply:
No, I suggest adding only `_before`. If other tests need to have some data for them to work they should do be explicit and not just relly that database is in a certain shape.

Reply:
But I see that we use `_after` to clean database on most other integration tests. Wouldn't using `_before` here be inconsistent ?

Reply:
Maybe inconsistent, but doing that has certain problems:
- if we clean in both `_after` and `_before` we are making our test runs slower 
- if we only delete in `_after` because we want to avoid conflicts we are creating hidden dependencies. Some tests have hidden dependencies. That means some tests rely on clean tables and people don't expect that and that makes tests harder to understand and harder to maintain. 
- when one test fails to delete its data (and that could be for whatever reason, bug, failure, ...) another test fails as well. But this second test fail is not because there is something wrong with the test or the code. It is just flaky.